File: src/main/java/io/github/shirohoo/library/domain/user/loanhistory/UserLoanHistory.java
package io.github.shirohoo.library.domain.user.loanhistory;

import static jakarta.persistence.GenerationType.IDENTITY;
import io.github.shirohoo.library.domain.user.User;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;

@Entity
public class UserLoanHistory {
    @Id
    @GeneratedValue(strategy = IDENTITY)
    private Long id;

    @ManyToOne
    private User user;

    private String bookName;

    private boolean isReturn;

    public UserLoanHistory() {

    }

    public UserLoanHistory(User user, String bookName, boolean isReturn) {
        this.user = user;
        this.bookName = bookName;
        this.isReturn = isReturn;
    }

    public String getBookName() {
        return this.bookName;
    }

    public void doReturn() {
        this.isReturn = true;
    }
}

File: src/main/kotlin/io/github/shirohoo/library/application/domain/user/UserLoanHistory.kt
package io.github.shirohoo.library.application.domain.user

import javax.persistence.*

@Entity
class UserLoanHistory(
    @ManyToOne
    val user: User,

    val bookTitle: String,

    isReturn: Boolean,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
) {
    var isReturn = isReturn
        private set

    fun doReturn() {
        this.isReturn = true
    }
}